application/src/main/java/org/thingsboard/server/service/install/update/DefaultDataUpdateService.java

```diff
@@ -328,6 +339,27 @@ private void updateNestedRuleChains() {
         }
     }
 
+    private void updateCheckRelationNode() {
+        PageDataIterable<RuleNode> ruleNodesIterator = new PageDataIterable<>(
+                link -> ruleChainService.findAllRuleNodesByType(TbCheckRelationNode.class.getName(), link), 1024);
+        ruleNodesIterator.forEach(ruleNode -> {
+            TbCheckRelationNodeConfiguration configNode = JacksonUtil.convertValue(ruleNode.getConfiguration(), TbCheckRelationNodeConfiguration.class);
```

Comment:
Ensure the update will not swap configuration multiple times if we execute upgrade script twice.